include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+ */
+int senss_set_sensitivity(
+		int handle,
+		int index, uint32_t value);
+
+/**
+ * @brief Get current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support get separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_get_sensitivity(
+		int handle,
+		int index, uint32_t *value);
+
+/**
+ * @brief Read a data sample
+ *
+ * This API will trigger target sensor read a sample each time.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer to receive sensor data sample.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_read_sample(
+		int handle,
+		void *buf, int size);
+
+/**
+ * @brief Sensor data event receive callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer with sensor data.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @param param The user private parameter.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+typedef int (*senss_data_event_t)(
+		int handle,
+		void *buf, int size, void *param);
+
+/**
+ * @brief Register sensor data event receive callback.
+ *
+ * User can use this API to register a data event receive callback function.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param callback The data event receive callback function.
+ *
+ * @param param The user private parameter.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_register_data_event_callback(
+		int handle,
+		senss_data_event_t callback, void *param);
+
+
+/**
+ * @brief Set report latency of a sensor instance.
+ *
+ * Report latency controls how long the sensor data events of the sensor instance will be buffered
+ * in the batching buffer before deliver to its client.
+ *
+ * For 0 latency value, it's means sensor data event will not buffered, and will report to its
+ * client immediately.
+ *
+ * The maximum sensor data events of this sensor instance is calculated as: latency /
+ * report interval.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param latency The latency in micro seconds.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_set_report_latency(
+		int handle, uint64_t latency);
+
+/**
+ * @brief Get current report latency value of a sensor instance.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param latency The value buffer to receive report latency in micro seconds.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_report_latency(
```

Comment:
for this and above, the `flush` function uses the term `batching` which is difficult to associate with `report`. Can we keep the wording the same and use `senss_(get|set)_batching_timeout` or something like that?

Reply:
how about `senss_batching_(get|set)_report_latency` or just `senss_batching_(get|set)_latency`? Since `report latency` is for control batching sample report latency, put all `batching` related APIs under `senss_batching_` prefix like `senss_batching_flush` is better.

Reply:
:+1: I like `senss_batching_(get|set)_latency` (though would prefer one API to atomically set 1+ attributes described in my latest comment on https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1145132339)

Reply:
removed this API since it's not used in PR1